test/shapirowilk.jl

```diff
@@ -0,0 +1,116 @@
+
+        swc10 = HypothesisTests.ShapiroWilkCoefs(10)
+        res = swc10.A .- [0.573715, 0.32897, 0.214349, 0.122791, 0.0400887]
+        @test norm(res, 1) ≈ 0.0 atol = length(swc10) * eps(Float32)
+
+        swc20 = HypothesisTests.ShapiroWilkCoefs(20)
+        res = swc20.A .-
+              [0.473371, 0.32174, 0.255663, 0.208297, 0.16864, 0.133584, 0.101474,
+               0.0712893, 0.0423232, 0.0140351]
+        @test norm(res, 1) ≈ 0.0 atol = length(swc20) * eps(Float32)
+
+        rng = StableRNG(0x5bca7c69b794f8ce)
+        X = sort(rand(rng, Float64, 10))
+        # W, pval, _ = swilkfort(X)
+        W = 0.9088434774710951
+        # pval = 0.2731410626084226
+        @test HypothesisTests.swstat(X, swc10) ≈ W atol = eps(Float32)
+    end
+
+    @testset "ShapiroWilk" begin
+        # syntactic tests
+        @test_throws ArgumentError ShapiroWilkTest([1, 2])
+        @test_throws ArgumentError ShapiroWilkTest([1, 2, 3], censored=4)
+        @test_throws DimensionMismatch ShapiroWilkTest([1, 2, 3],
+                                                       HypothesisTests.ShapiroWilkCoefs(4))
+
+        t = ShapiroWilkTest([1, 2, 3])
+        @test t.W == 1.0
+        @test HypothesisTests.pvalue(t) == 1.0
+
+        str = sprint(show, t)
+        @test str ==
+              """Shapiro-Wilk normality test
+              ---------------------------
+              Population details:
+                  parameter of interest:   Squared correlation of data and the expected order statistics of N(0,1) (W)
+                  value under h_0:         1.0
+                  point estimate:          1.0
+
+              Test summary:
+                  outcome with 95% confidence: fail to reject h_0
+                  one-sided p-value:           1.0000
+
+              Details:
+                  number of observations: 3
+                  censored ratio:         0.0
+                  W-statistic:            1.0
+              """
+
+        # testing different cases of N
+        for N in (3, 5, 11, 12)
+            rng = StableRNG(0x5bca7c69b794f8ce)
+            X = sort(randn(rng, N))
+            t = ShapiroWilkTest(X; sorted=true)
+
+            # analytic properties from Shapiro-Wilk 1965:
+            # Lemma 1: Scale and origin invariance:
+            scale, shift = rand(rng, 2)
+            @test t.W ≈ ShapiroWilkTest(X .+ shift).W
+            @test t.W ≈ ShapiroWilkTest(scale .* X).W
+            @test t.W ≈ ShapiroWilkTest(scale .* X .+ shift).W
+            # Lemma 2, 3: upper and lower bounds
+            @test N * t.coefs[1]^2 / (N - 1) ≤ t.W ≤ 1.0
+        end
+
+        @testset "Worked Example" begin
+            # **Worked Example** (Section 4) from
+            # PATRICK ROYSTON Approximating the Shapiro-Wilk W-test for non-normality
+            # *Statistics and Computing* (1992) **2**, 117-119
+
+            X = [48.4, 49.0, 59.5, 59.6, 60.7, 88.8, 98.2, 109.4, 169.1, 227.1]
```

Comment:
Let's avoid sorting values so that we check that we sort values when they aren't. Actually it would be good to compare the results with and without `sorted=true`.
```suggestion
            X = [169.1, 109.4, 49.0, 59.5, 59.6, 60.7, 98.2, 227.1, 48.4, 88.8]
```

Reply:
I added a few other tests (including the permutation of this sample)